From 379d967e516f920faff9ab052b1cf2e9d32012b2 Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Wed, 1 Apr 2015 10:31:33 +0800
Subject: [PATCH] MLK-10525:DTS: Move sii902x hdmi property to standalone dts

imx7D SDB bootup will hang at sii902x HDMI driver loading
if imx7d sdb didn't HW rework pull up sii902x DDC to 3.3V
via 2K resistor.
Before board team done the DDC rework for all imx7D SDB board,
sii902x driver move to standalone dts.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 arch/arm/boot/dts/Makefile           |  3 ++-
 arch/arm/boot/dts/imx7d-sdb-hdmi.dts | 16 ++++++++++++++++
 arch/arm/boot/dts/imx7d-sdb.dts      |  2 +-
 3 files changed, 19 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index a154807..71f151a 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -220,7 +220,8 @@ dtb-$(CONFIG_ARCH_MXC) += \
 	imx7d-sdb.dtb \
 	imx7d-sdb-enet.dtb \
 	imx7d-sdb-gpmi-weim.dtb \
-	imx7d-sdb-epdc.dtb
+	imx7d-sdb-epdc.dtb \
+	imx7d-sdb-hdmi.dtb
 dtb-$(CONFIG_ARCH_MXS) += imx23-evk.dtb \
 	imx23-olinuxino.dtb \
 	imx23-stmp378x_devb.dtb \
diff --git a/arch/arm/boot/dts/imx7d-sdb-hdmi.dts b/arch/arm/boot/dts/imx7d-sdb-hdmi.dts
new file mode 100644
index 0000000..7b4aa27
--- /dev/null
+++ b/arch/arm/boot/dts/imx7d-sdb-hdmi.dts
@@ -0,0 +1,16 @@
+/*
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx7d-sdb.dts"
+
+&i2c3 {
+	sii902x@39 {
+		status = "okay";
+	};
+};
+
diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index 768ef23..a627c96 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -375,7 +375,7 @@
 		mode_str ="1280x720M@60";
 		bits-per-pixel = <16>;
 		reg = <0x39>;
-		status = "okay";
+		status = "disabled";
 	};
 };
 
-- 
1.8.0

